controlplane/kubeadm/internal/workload_cluster_test.go

```diff
@@ -126,57 +161,92 @@ func TestUpdateKubeProxyImageInfo(t *testing.T) {
-			clientGet: map[string]interface{}{
-				"kube-system/" + kubeProxyDaemonSetName: ds,
-			},
+			name:        "succeeds if patch correctly",
+			ds:          newKubeProxyDS(),
+			expectErr:   false,
+			expectImage: "k8s.gcr.io/kube-proxy:v1.16.3",
+			KCP:         &v1alpha3.KubeadmControlPlane{Spec: v1alpha3.KubeadmControlPlaneSpec{Version: "v1.16.3"}},
 		},
 		{
-			name:      "returns error if image in kube-proxy ds was in digest format",
-			ds:        dsImageInDigestFormat,
-			expectErr: true,
-			clientGet: map[string]interface{}{
-				"kube-system/" + kubeProxyDaemonSetName: dsImageInDigestFormat,
-			},
+			name:        "returns error if image in kube-proxy ds was in digest format",
+			ds:          newKubeProxyDSWithImage("k8s.gcr.io/kube-proxy@sha256:47bfd"),
+			expectErr:   true,
+			expectImage: "k8s.gcr.io/kube-proxy@sha256:47bfd",
+			KCP:         &v1alpha3.KubeadmControlPlane{Spec: v1alpha3.KubeadmControlPlaneSpec{Version: "v1.16.3"}},
+		},
+		{
+			name:        "expects v-prefixed format of tag",
+			ds:          newKubeProxyDS(),
+			expectErr:   false,
+			expectImage: "k8s.gcr.io/kube-proxy:v1.16.3",
+			KCP:         &v1alpha3.KubeadmControlPlane{Spec: v1alpha3.KubeadmControlPlaneSpec{Version: "1.16.3"}},
+		},
+		{
+			name:        "expects OCI compatible format of tag",
+			ds:          newKubeProxyDS(),
+			expectErr:   false,
+			expectImage: "k8s.gcr.io/kube-proxy:v1.16.3_build1",
+			KCP:         &v1alpha3.KubeadmControlPlane{Spec: v1alpha3.KubeadmControlPlaneSpec{Version: "v1.16.3+build1"}},
 		},
 		{
 			name:      "returns error if image in kube-proxy ds was in wrong format",
-			ds:        ds,
+			ds:        newKubeProxyDSWithImage(""),
 			expectErr: true,
-			clientGet: map[string]interface{}{
-				"kube-system/" + kubeProxyDaemonSetName: dsImageEmpty,
-			},
+			KCP:       &v1alpha3.KubeadmControlPlane{Spec: v1alpha3.KubeadmControlPlaneSpec{Version: "v1.16.3"}},
 		},
 	}
 
 	ctx := context.Background()
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-
-			fakeClient := &fakeClient{
-				get: tt.clientGet,
+			objects := []runtime.Object{
+				&tt.ds,
 			}
-			c := &Workload{
+			fakeClient := fake.NewFakeClientWithScheme(scheme, objects...)
+			w := &Workload{
 				Client: fakeClient,
 			}
-			err := c.UpdateKubeProxyImageInfo(ctx, &v1alpha3.KubeadmControlPlane{Spec: v1alpha3.KubeadmControlPlaneSpec{Version: "v1.16.3"}})
+			err := w.UpdateKubeProxyImageInfo(ctx, tt.KCP)
 			if err != nil && !tt.expectErr {
 				t.Fatalf("expected no error, got %s", err)
 			}
 			if err == nil && tt.expectErr {
 				t.Fatal("expected error but got none")
 			}
-
+			proxyImage, err := w.getProxyImageInfo(ctx)
+			if err != nil {
+				t.Fatalf("expected no error, got %s", err)
+			}
+			if proxyImage != tt.expectImage {
+				t.Fatalf("expected proxy image %s, got %s", tt.expectImage, proxyImage)
+			}
 		})
 	}
 }
+
+func (w *Workload) getProxyImageInfo(ctx context.Context) (string, error) {
```

Comment:
Is this method just for tests? 
If so, I recommend using the `fakeClient` to get the object information required for assertions.
If this is going to be used in other methods then maybe move it to `workload_cluster.go`.

Reply:
might want to rewrite it as a function